test/utils/utils.go

```diff
@@ -247,3 +256,67 @@ func UncommentCode(filename, target, prefix string) error {
 	// nolint:gosec
 	return os.WriteFile(filename, out.Bytes(), 0644)
 }
+
+func LoadContainerImageToKindCluster(kindCluster *kind.Cluster,
+	ctx context.Context,
+	dockerImage string,
+	containerTool string) (context.Context, error) {
+	if containerTool == "podman" {
+		if ctx, err := useImageArchive(kindCluster, ctx); err != nil {
+			log.Println(err)
+			return ctx, err
+		}
+	} else {
+		if err := kindCluster.LoadImage(ctx, dockerImage); err != nil {
+			log.Printf("Failed to load image into kind: %s", err)
+			return ctx, err
+		}
+	}
+	return ctx, nil
+}
+
+func useImageArchive(kindCluster *kind.Cluster, ctx context.Context) (context.Context, error) {
+	var err error
+	dname, err := os.MkdirTemp("", "etcdOperator")
+	if err != nil {
+		return ctx, nil
+	}
+
+	fname := filepath.Join(dname, archiveName)
+
+	defer func() {
+		if err := os.RemoveAll(fname); err != nil {
+			log.Printf("failed to remove %s: %s", fname, err)
+		}
+	}()
+
+	log.Println("Creating docker image archive...")
+	if err := createImageArchive(fname); err != nil {
+		log.Printf("Failed to create archive %s:", err)
+		return ctx, err
+	}
+
+	log.Println("Loading docker image archive...")
+	ctx, err = loadImageArchive(*kindCluster, ctx, fname)
+	if err != nil {
+		return ctx, err
+	}
+	return ctx, nil
+}
+
+func createImageArchive(archivePath string) error {
+	// Save docker image archive
+	cmd := exec.Command("make", "docker-save", fmt.Sprintf("ARCHIVE_NAME=%s", archivePath))
+	if _, err := Run(cmd); err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadImageArchive(kindCluster kind.Cluster, ctx context.Context, archivePath string) (context.Context, error) {
+	// Load archive into cluster
+	if err := kindCluster.LoadImageArchive(ctx, archivePath); err != nil {
+		return ctx, err
+	}
+	return ctx, nil
+}
```

Comment:
Minor comment: These two functions do not make much sense. I would suggest to move the implementation into `useImageArchive` directly.